readthedocs/doc_builder/director.py

```diff
@@ -622,17 +623,28 @@ def install_build_tools(self):
                 )
                 # If the tool version selected is not available from the
                 # cache we compile it at build time
-                cmd = [
-                    # TODO: make ``PYTHON_CONFIGURE_OPTS="--enable-shared"``
-                    # environment variable to work here. Note that
-                    # ``self.build_environment.run`` does not support passing
-                    # environment for a particular command:
-                    # https://github.com/readthedocs/readthedocs.org/blob/9d2d1a2/readthedocs/doc_builder/environments.py#L430-L431
-                    "asdf",
-                    "install",
-                    tool,
-                    full_version,
-                ]
+                
+                # For miniforge3, unset CONDA_ENVS_PATH and CONDA_DEFAULT_ENV
+                # environment variables to allow proper installation
+                # See https://github.com/readthedocs/readthedocs.org/issues/11690
+                if tool == "python" and full_version.startswith("miniforge3"):
+                    cmd = [
+                        "/bin/sh",
+                        "-c",
+                        f"unset CONDA_ENVS_PATH ; unset CONDA_DEFAULT_ENV ; asdf install {shlex.quote(tool)} {shlex.quote(full_version)}",
+                    ]
```

Comment:
This probably works, but I don't like it.

I'd like to make this change in `get_build_env_vars` instead. It will be cleaner there.

...

Hrm, I'm not sure that's possible because we use that method to initialize the `build_environment`. Another option was to call `build_environment.run(..., environment=)` but we block that on purpose in https://github.com/readthedocs/readthedocs.org/blob/main/readthedocs/doc_builder/environments.py#L530-L534

Is there a better option here? 🤔

Reply:
I just realized that we don't need to unset these variables because we pre-compile these tools, so there is no need to call `asdf install python` command 👍🏼